sklearn/tests/test_calibration.py

```diff
@@ -341,3 +342,21 @@ def decision_function(self, X):
     calibrated_clf = CalibratedClassifierCV(MockTensorClassifier())
     # we should be able to fit this classifier with no error
     calibrated_clf.fit(X, y)
+
+
+def test_calibration_with_fit_params():
+    """Tests that fit_params are passed to the underlying base estimator
+
+    https://github.com/scikit-learn/scikit-learn/issues/12384
+
+    """
+    n_samples = 100
+    X, y = make_classification(n_samples=2 * n_samples, n_features=6,
+                               random_state=42)
+
+    fit_params = {'a': y, 'b': y}
```

Comment:
Please also check with non-arrays, such as lists

Reply:
What would be the expectation here? That lists work as long as they have the correct length? Is there something else that should work?

Reply:
Yes, lists are the main thing to check, under the assumption that if they are handled, so are other sequences

Reply:
With lists, the line `fit_params_train[k] = v[train]` fails, as expected. I thus wanted to use `indexable` to make sure that lists would pass but actually, `indexable` doesn't touch lists because of this line:

https://github.com/scikit-learn/scikit-learn/blob/1495f69242646d239d89a5713982946b8ffcf9d9/sklearn/utils/validation.py#L224-L225

What is the right way to deal with lists in this case?

Reply:
So I think you need to use safe_indexing rather than `v[train]`

Reply:
Thx, I forgot about that function. I used `_safe_indexing` instead of `safe_indexing` because the latter is deprecated.